src/main/scala/scalaz/stream/wye.scala

```diff
@@ -836,9 +836,7 @@ object wye {
         }
       })(S)
 
-      repeatEval(Task.async[Seq[O]] { cb => a ! Get(cb) })
-      .flatMap(emitAll)
-      .onComplete(eval_(Task.async[Unit](cb => a ! DownDone(cb))))
+      repeatEval(Task.async[Seq[O]] { cb => a ! Get(cb) }) onHalt { _.asHalt } flatMap emitAll onComplete eval_(Task.async[Unit](cb => a ! DownDone(cb)))
```

Comment:
There are several reasons *not* to use dot notation, most notably that it rules out using curly braces, which in turn rules out semicolons and generates inconsistencies in newline inference.  I would certainly be willing to split the line up, because it really is too long as you point out, but I feel very very strongly about the demerits of dot notation in Scala.  Infix notation is quite consistent in Scala, and in fact it is one of the only ways to generate a method invocation and block syntax which is consistent across all constructs.